Test/ElasticScale.ShardManagement.UnitTests/ScenarioTests.cs

```diff
@@ -1237,6 +1237,119 @@ public void BasicScenarioRangeShardMaps()
+                ShardLocation sl2 = new ShardLocation(
+                        Globals.ShardMapManagerTestsDatasourceName,
+                        ScenarioTests.s_perTenantDBs[1]);
+
+                Shard s2 = lsm.CreateShard(sl2);
+
+                PointMapping<int> p2 = lsm.CreatePointMapping(2, s2);
+                Assert.IsTrue(ValidateCounterValue(instanceName, PerformanceCounterName.MappingsCount, 2));
+
+                // perform DDR operation few times and validate non-zero counter values
+                for (int i = 0; i < 10; i++)
+                {
+                    using (SqlConnection conn = lsm.OpenConnectionForKey(1, Globals.ShardUserConnectionString))
+                    {
+                    }
+                }
+
+                Assert.IsTrue(ValidateNonZeroCounterValue(instanceName, PerformanceCounterName.DdrOperationsPerSec));
+                Assert.IsTrue(ValidateNonZeroCounterValue(instanceName, PerformanceCounterName.MappingsLookupSucceededPerSec));
+
+                // Remove shard map after removing mappings and shard
+                lsm.DeleteMapping(lsm.MarkMappingOffline(lsm.GetMappingForKey(1)));
+                lsm.DeleteMapping(lsm.MarkMappingOffline(lsm.GetMappingForKey(2)));
+                lsm.DeleteShard(lsm.GetShard(sl1));
+                lsm.DeleteShard(lsm.GetShard(sl2));
+
+                Assert.IsTrue(ValidateCounterValue(instanceName, PerformanceCounterName.MappingsCount, 0));
+
+                smm.DeleteShardMap(lsm);
+
+                // make sure that perf counter instance is removed
+                Assert.IsFalse(ValidateInstance(instanceName));
+            }
+        }
+
+        private bool ValidateNonZeroCounterValue(string instanceName, PerformanceCounterName counterName)
+        {
+            string counterdisplayName = (from c in PerfCounterInstance.counterList
+                                         where c.CounterName == counterName
+                                         select c.CounterDisplayName).First();
+
+            using (PerformanceCounter pc =
+                new PerformanceCounter(PerformanceCounters.ShardManagementPerformanceCounterCategory, counterdisplayName, instanceName))
+            {
+                return pc.RawValue != 0;
+            }
+        }
+
+        private bool ValidateCounterValue(string instanceName, PerformanceCounterName counterName, long value)
+        {
+            string counterdisplayName = (from c in PerfCounterInstance.counterList
+                                         where c.CounterName == counterName
+                                         select c.CounterDisplayName).First();
+
+            using (PerformanceCounter pc =
+                new PerformanceCounter(PerformanceCounters.ShardManagementPerformanceCounterCategory, counterdisplayName, instanceName))
+            {
+                return pc.RawValue.Equals(value);
+            }
+        }
+
+        private bool ValidateInstance(string instanceName)
```

Comment:
Renaming it to ValidateInstanceExists since this method is only validating instance and not looking at any counter data. Avoiding confusion with another method which checks counter existence - PerformanceCounterCategory.CounterExists().